src/app/js/lib/selectors.ts

```diff
@@ -9,12 +10,33 @@ export const createGlobalSelector = (getLocalState, selector) =>
         selector(localState, props),
     );
 
-// @ts-expect-error TS7006
-export const createGlobalSelectors = (getLocalState, selectors) =>
+// export const createGlobalSelectors = <
+//     T extends Record<string, (state: any) => any>,
+// >(
+//     getLocalState,
+//     selectors,
+// ) =>
+//     Object.keys(selectors).reduce(
+//         (result, key) => ({
+//             ...result,
+//             [key]: createGlobalSelector(getLocalState, selectors[key]),
+//         }),
+//         {},
+//     );
+export const createGlobalSelectors = <
+    T extends Record<string, (localState: any, props?: any) => any>,
```

Comment:
What about something like that? I think it would improve type checking.

```suggestion
    T extends Record<string, (localState: LocalState, props?: Props) => any>,
```

Reply:
It's too soon to type the global state
We first need to do all sub reducer

Reply:
It has too many side effect, we need to left some part untyped, otherwise it never ends